.appveyor.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+set -eo pipefail
+
+BUILD_JDK_URL='https://github.com/AdoptOpenJDK/openjdk12-binaries/releases/download/jdk-12.0.2%2B10/OpenJDK12U-jdk_x64_linux_hotspot_12.0.2_10.tar.gz'
+JRE8_URL='https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u222-b10/OpenJDK8U-jre_x64_linux_hotspot_8u222b10.tar.gz'
+JRE11_URL='https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.4%2B11/OpenJDK11U-jre_x64_linux_hotspot_11.0.4_11.tar.gz'
+JRE13_URL='https://github.com/AdoptOpenJDK/openjdk13-binaries/releases/download/jdk-13%2B33/OpenJDK13U-jre_x64_linux_hotspot_13_33.tar.gz'
+
+function msg() {
+  echo -ne "\033[1;32m"
+  echo -n "$@"
+  echo -e "\033[0m"
+}
+
+function echo_and_run() {
+  msg "$@"
+  "$@"
+}
+
+if [[ $# -ne 1 ]]; then
+  echo "Usage: $0 <profile>" >&2
+  exit 1
+fi
+
+# Prepare the environment variables based on the specified profile.
+PROFILE="$1"
+case "$PROFILE" in
+java8)
+  TEST_JRE_URL="$JRE8_URL"
+  TEST_JAVA_VERSION='8'
+  COVERAGE=0
+  ;;
+java11)
+  TEST_JRE_URL="$JRE11_URL"
+  TEST_JAVA_VERSION='11'
+  COVERAGE=1
+  ;;
+java13)
+  TEST_JRE_URL="$JRE13_URL"
+  TEST_JAVA_VERSION='13'
+  COVERAGE=0
+  ;;
+*)
+  echo "Unknown profile: $PROFILE" >&2
+  exit 1
+  ;;
+esac
+
+export TEST_JAVA_VERSION
+export JAVA_HOME="$HOME/jdk/build"
+export JAVA_TEST_HOME="$HOME/jdk/test-$TEST_JAVA_VERSION"
+export PATH="$JAVA_HOME/bin:$PATH"
+
+# Restore the home directory from the cache if necessary.
+if [[ -d /var/cache/appveyor ]] && \
+   [[ -n "$APPVEYOR_ACCOUNT_NAME" ]] && \
+   [[ -n "$APPVEYOR_PROJECT_SLUG" ]] && \
+   [[ -n "$APPVEYOR_REPO_BRANCH" ]]; then
+
+  # Purge cache directories not touched for last 7 days.
+  find /var/cache/appveyor -mindepth 4 -maxdepth 4 -type d -mtime +7 \
+      -exec echo 'Purging:' {} ';' \
+      -exec rm -fr {} ';'
+  # Delete the empty directories.
+  find /var/cache/appveyor -mindepth 1 -maxdepth 4 -type d -empty \
+      -exec echo 'Purging:' {} ';' \
+      -delete
+
+  if [[ "$PURGE_CACHE" != '1' ]]; then
+    # Restore the home directory from the cache.
+    BRANCH_CACHE_DIR="/var/cache/appveyor/$APPVEYOR_ACCOUNT_NAME/$APPVEYOR_PROJECT_SLUG/branches/$APPVEYOR_REPO_BRANCH"
+    if [[ -z "$APPVEYOR_PULL_REQUEST_NUMBER" ]]; then
+      CACHE_DIR="$BRANCH_CACHE_DIR"
+    else
+      CACHE_DIR="/var/cache/appveyor/$APPVEYOR_ACCOUNT_NAME/$APPVEYOR_PROJECT_SLUG/pulls/$APPVEYOR_PULL_REQUEST_NUMBER"
```

Comment:
We should use the same cache for all the builds. If it's like this I think every build is slow unless it is a follow up commit in a PR.

Reply:
If CACHE_DIR does not exist, it will use BRANCH_CACHE_DIR as a seed, so it should be fine once we merge this PR and the cache for master branch is built, no?